ports/stm32/Makefile

```diff
@@ -63,12 +63,17 @@ INC += -Ilwip_inc
 # Basic Cortex-M flags
 CFLAGS_CORTEX_M = -mthumb
 
+SUPPORTS_HARDWARE_FP_SINGLE = 1
+SUPPORTS_HARDWARE_FP_DOUBLE = 0
```

Comment:
I'm pretty sure you can't override make variables, so these should use `?=`...

... but IMO it'd be cleaner to define both these variables in the if-logic below.  Yes there would then be some minor duplication of lines but I think it would be clearer.